app/reviews/create/page.tsx

```diff
@@ -1,77 +1,172 @@
-'use client'
+"use client";
 
-// TEST PAGE FOR CREATE ACTION - PLEASE REPLACE
-import { useFormState } from 'react-dom'
+import { useFormState } from "react-dom";
 import { createReview } from "./actions";
+import { NextPage } from "next";
+import { createClient } from "@/utils/supabase/client";
 
-const createReviewPage = () => {
-  // define example propertyId and address, these should be determined by params in implementation
-  const examplePropertyId = undefined // '7cda6a4c-868d-436d-a05b-25d6eeaed351'
-  const exampleAddress = "1234 Example St, Example, EX 12345";
+const createReviewPage: NextPage = async ({
+  searchParams
+}: {
+  searchParams?: {
+    propertyId?: string,
+    address?: string,
+  }
+}) => {
+  const propertyId = searchParams?.propertyId;
+  let address: string|undefined = undefined;
+
+  // get the address from the propertyId if it's defined
+  if (propertyId) {
+    const supabase = createClient();
+
+    const { data, error } = await supabase
+      .from("properties")
+      .select("address")
+      .eq("id", propertyId)
+      .maybeSingle();
+    
+    // Consider adding error handling here
+    // redirect to / show not-found if no property found with the provided id
+
+    address = data?.address;
+  }
+
+  if (!address) address = searchParams?.address /* For Testing, add dummy value if none provided, consider replacing with error handling as above */ || "1 Lorem Ipsum Road, Oldfield Park";
 
   // bind the id & address to the createReview function
   const initialState = { message: null, errors: {} };
-  const createReviewWithProperty = createReview.bind(null, {id: examplePropertyId, address: exampleAddress});
-  const [state, dispatch] = useFormState(createReviewWithProperty, initialState)
+  const createReviewWithProperty = createReview.bind(null, {
+    id: propertyId,
+    address: address,
+  });
+  const [state, dispatch] = useFormState(
+    createReviewWithProperty,
+    initialState
+  );
 
   return (
-    <div>
-      <h1>Create Review</h1>
-      <p>Test page for create review action - REPLACE ME</p>
+    <div style={{ minWidth: "1000px" }}>
+      <h1 className="text-4xl font-bold mb-2 mt-8">Add New Review</h1>
+      <p className="mb-8 text-xl">
+        Add a new review for this property:{" "}
+        <span>{address}</span>
+      </p>
 
       <form
-        className="flex flex-col gap-2 pleace-items-center max-w-prose w-full border p-4 rounded-md"
+        className="flex flex-col gap-2 pleace-items-center w-full border p-4 rounded-md mb-8"
         action={dispatch}
+        style={{ backgroundColor: "#778378", color: "#222" }}
```

Comment:
are these colours supposed to cover the whole page as in the designs?

we probably want to [add them as custom colours in the tailwind theme](https://tailwindcss.com/docs/customizing-colors#adding-additional-colors)

and then set the background in the root layout so that it covers the whole page, as I assume you do not want it to look like this: 

<img width="1465" alt="Screenshot 2024-02-17 at 22 01 00" src="https://github.com/CM20254-Group-7/Rental_Review/assets/74842299/186d70d1-3bb5-4b81-b23c-8c95e9f0a057">

Reply:
So when I tried applying the color to the whole div, it applied the color only to the small column occupied by the div rather than the whole page. So as you say we should change it in the root layout if we agree all pages should have this background color. 

It looked very weird when just the div had the background color, so I kept it only on the form for now, but later when we change the color of the whole page we should change the bg-color to the whole div rather than just the form. I'll change the style from just the form to the whole div in the next commit.

Reply:
I'm pretty sure we have it as the background colour in all the prototype mockups, so it would make sense to set in the root layout.

If these are standard colours, it would also make sense to use css variables for them. See[here](https://github.com/CM20254-Group-7/Rental_Review/blob/c0716203d141e149900cede1780e7503683e9b5e/app/globals.css#L5-L21) for setting variables for the light and dark theme and [here](https://github.com/CM20254-Group-7/Rental_Review/blob/c0716203d141e149900cede1780e7503683e9b5e/tailwind.config.js#L9-L16) for extending tailwind with those variables.

You can either replace the existing background variable values, or create a one e.g. 'primary'
 
This is potentially worth adding a new task as part of a 'consistent UI design' story for especially since it will involve going over each existing page and ensuring compatability

Reply:
You might want to have a look at something like realtime colours. Which can export themes in this format:

<img width="1261" alt="Screenshot 2024-02-20 at 14 39 57" src="https://github.com/CM20254-Group-7/Rental_Review/assets/74842299/275f04fe-795d-4c1e-93b0-8eaad701f9ef">

Have a look at this [example I made with mostly colours from the prototypes](https://www.realtimecolors.com/?colors=212121-778378-dbb866-ededed-443dff&fonts=Poppins-Poppins)

Reply:
dealing with this in separate tasks